### PR TITLE
Testing current behavior of if/then/else

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Related to #9165 work. Let's make sure the current behavior of `if`/`then`/`else` is properly tested by a unit test. Extracting test created as part of #11365 to verify it really describes the current behavior.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
- [x] Unit tests have been written where possible.

### CODE REVIEW

**Review by 4e6 — APPROVED** [review]

File: engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class IfThenElseTest {
+  private static Context ctx;
+  private static final ByteArrayOutputStream MESSAGES = new ByteArrayOutputStream();
+
+  @BeforeClass
+  public static void initEnsoContext() {
+    ctx =
```
Comment (Akirathan) [inline engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java]: DRY: Use [`ContextUtils.defaultCOntextBuilder`](https://github.com/enso-org/enso/blob/fb821303ab2818c4ce2d8482612584497f2e2014/lib/java/test-utils/src/main/java/org/enso/test/utils/ContextUtils.java#L42).

File: engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    ctx = null;
+  }
+
+  @Test
+  public void simpleIfThenElse() throws Exception {
+    var module = ctx.eval("enso", """
+    check x = if x then "Yes" else "No"
+    """);
+
+    var check = module.invokeMember(MethodNames.Module.EVAL_EXPRESSION, "check");
```
Comment (Akirathan) [inline engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java]: DRY: Use [`ContextUtils.getMethodFromModule`](https://github.com/enso-org/enso/blob/fb821303ab2818c4ce2d8482612584497f2e2014/lib/java/test-utils/src/main/java/org/enso/test/utils/ContextUtils.java#L204)

**Review by Akirathan — APPROVED** [review]
DRY nits

File: engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java (hunk 2 above)
Comment (JaroslavTulach) [reply engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java]: Done in fc17113

File: engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java]: Done in fc1711371b3017cd3ccd873e2b4b486cf262b2ea

**Comment by JaroslavTulach** [comment]
Expanded the list of tests to cover warnings. Also to use `BoolObject extends TruffleObject` interop - surprisingly that one is [broken](https://github.com/enso-org/enso/pull/11395/commits/d9be3bb8ef51614a83a4cde14b00288aaf0cb56a) and needs to be `@Ignore`d for now.

**Pull request merged**